extensions/podman/src/extension.ts

```diff
@@ -108,11 +118,26 @@ export type MachineInfo = {
 
 async function updateMachines(provider: extensionApi.Provider): Promise<void> {
   // init machines available
-  const machineListOutput = await getJSONMachineList();
+  let machineListOutput: string;
+  try {
+    machineListOutput = await getJSONMachineList();
```

Comment:
I find it weird that we do a try / catch around this?

Shouldn't we keep it like:
```ts
const machineListOutput = await getJSONMachineList();
if (machineListOutput === undefined) {
    if (shouldNotifySetup) {
      // push setup notification
      extensionApi.window.showNotification(setupPodmanNotification);
      shouldNotifySetup = false;
    }
}

?

I find it odd we also throw an error but it doesn't appear anywhere?

Reply:
try / catch also prevents the rest of the activate() to proceed?

Reply:
the problem is that if you don't put the `getJSONMachineList` inside a try/catch and it fails for some reason (like if it tries to fetch machines when podman is not installed) you never display the notification. This way if there is an error, it checks if the notification needs to be shown and throw the error as it would have done before.